src/library/columncache.h

```diff
@@ -110,7 +110,8 @@ class ColumnCache : public QObject {
         if (index < 0) {
             return;
         }
-        DEBUG_ASSERT(!m_columnSortByIndex.contains(index));
+        DEBUG_ASSERT(!m_columnSortByIndex.contains(index) ||
+                m_columnSortByIndex[index] == sortFormat);
```

Comment:
We clear the m_columnSortByIndex before refilling it.
The original debug assertion asserts that no value is listed twice.
Now it is allowed to add the same value twice. I can imagine that this can happen when the clear() call got lost. 
Isn't this an issue we want to track?

Reply:
The original assertion allowed to insert the same value twice, but not different values. We apparently write different values for the same index which is unexpected. This did not happen before.

Reply:
Index clashes: Both COLUMN_LIBRARYTABLE_ARTIST and COLUMN_PLAYLISTTRACKSTABLE_ARTIST are mapped to index 6.

So we have still issue with ColumnCache that I once tried to demonstrate. The foundations are still broken.

Reply:
COLUMN_LIBRARYTABLE_ARTIST: index = 6, sortName = ""lower(%1) COLLATE mixxxLexicographicalCollationFunc"
COLUMN_PLAYLISTTRACKSTABLE_ARTIST: index = 6, sortName = ""lower(%1)"

Reply:
I am confused. Isn't this the case the assert id for? 
Does the assert strike right now?

Reply:
The restored debug asserts for this case, yes. But it still failed, because we map different columns to the same index. See the workaround that fixes the inconsistent values.

Reply:
I cannot confirm this. Current 2.3 runs without assertion in my case. 
What did you do to hit that assertion?

Reply:
main aborts. There are more debug assertion for various COs when creating a new profile during the first startup. The debug assertions regarding ColumnCache occur on every startup.

Reply:
```
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 21 "Channels"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 19 "Samplerate"
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 4 "Album Artist"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 21 "Channels"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 32 "Color"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 7 "Composer"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 33 "Cover Art"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 23 "Date Added"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 40 "Last Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 10 "Type"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 8 "Grouping"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 28 "Key"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 16 "ReplayGain"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 19 "Samplerate"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 25 "Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 43 "#"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 48 "Timestamp"
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 21 "Channels"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 32 "Color"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 7 "Composer"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 33 "Cover Art"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 23 "Date Added"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 40 "Last Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 10 "Type"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 28 "Key"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 16 "ReplayGain"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 19 "Samplerate"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 25 "Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 43 "#"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 48 "Timestamp"
Debug [Main] Traktor Library Location=[ "/home/uk/collection.nml" ]
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 4 "Album Artist"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 21 "Channels"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 32 "Color"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 7 "Composer"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 33 "Cover Art"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 23 "Date Added"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 40 "Last Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 10 "Type"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 8 "Grouping"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 16 "ReplayGain"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 19 "Samplerate"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 25 "Played"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 43 "#"
Debug [Main] BaseTrackTableModel - Skipping header properties for unsupported column 48 "Timestamp"
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] Dropping Rekordbox table:  "rekordbox_playlist_tracks"
Debug [Main] Dropping Rekordbox table:  "rekordbox_playlists"
Debug [Main] Dropping Rekordbox table:  "rekordbox_library"
Debug [Main] Creating Rekordbox library table:  "rekordbox_library"
Debug [Main] Creating Rekordbox playlists table:  "rekordbox_playlists"
Debug [Main] Creating Rekordbox playlist tracks table:  "rekordbox_playlist_tracks"
Debug [Main] Committing transaction successfully on "MIXXX-1"
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] Dropping Serato table:  "serato_playlist_tracks"
Debug [Main] Dropping Serato table:  "serato_playlists"
Debug [Main] Dropping Serato table:  "serato_library"
Debug [Main] Creating Serato library table:  "serato_library"
Debug [Main] Creating Serato playlists table:  "serato_playlists"
Debug [Main] Creating Serato playlist tracks table:  "serato_playlist_tracks"
Debug [Main] Committing transaction successfully on "MIXXX-1"
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Critical [Main] DEBUG ASSERT: "!m_columnSortByIndex.contains(index) || m_columnSortByIndex[index] == sortFormat" in function void ColumnCache::insertColumnSortByEnum(ColumnCache::Column, const QString&) at ../src/library/columncache.h:115
Debug [Main] aoide TrackTableModel - Created instance aoide::TrackTableModel(0x3f21290)
Debug [Main] aoide LibraryFeature - Created instance aoide::LibraryFeature(0x4138e10)
```

Reply:
I am now able to reproduce on 2.3 and main. I musse have messed up something during my initial tests. 

However, the ASSERT works. It has discovered a real bug. I will file a fix PR against this PR

Reply:
Why? This PR fixes the urgent issues. Everything else can be done in a separate PR. I don't want to extend the scope of this hotfix!

Reply:
Yes, right, lets merge this now.